TrigScint/src/TrigScint/EventReadoutProducer.cxx

```diff
@@ -89,38 +89,42 @@ void EventReadoutProducer::produce(framework::Event &event) {
     // normalization the same numbers are repeated, it's an oscillation edge
     // case: multiple single PE peaks with that repetition. we probably don't
     // need to keep those anyway
-    if (verbose_) ldmx_log(debug) << "going into oscillations check ";
     std::vector<float> chargeCheck = {NULL};
     float minCharge =
         10;  // no point in looking at oscillations just around the pedestal.
              // nearest edge is 10.35 fC  //ADC=0 corresponds to -16 fC.
-    for (int i = 3; i < charge.size() - 4; i++) {
-      float maxSamp = minCharge;
-      for (int iQ = 0; iQ < 4; iQ++) {  // find the local max in the 4 samples
-        //		ldmx_log(debug) << "got charge " << charge[i+iQ];
-        if (charge[i + iQ] > maxSamp) maxSamp = charge[i + iQ];
-      }
-      if (verbose_) ldmx_log(debug) << "got max charge " << maxSamp;
-      for (int iQ = 0; iQ < 4;
-           iQ++)  // store the locally normalized numbers. even if the period is
-                  // 5 this should work for a while
-        chargeCheck.push_back(charge[i + iQ] / maxSamp);
-      i += 3;  // to increment by 4, do 3 here and 1 in the loop
-    }
-
-    // now calculate the pedestal as the average of the middle half of the
-    // sorted vector
     float ped = 0;
-    std::sort(charge.begin(), charge.end());
     int pedLength = (int)charge.size() /
                     5;  // actually pulse can be up to 12 samples = 2/5*30
     int pedOffset = 2;  // but still skip the lowest (and highest) few
-    //	for (int i = pedLength; i < 3*pedLength ; i++) { //use 2nd and third 5th
-    for (int i = pedOffset; i < 2 * pedLength + pedOffset;
-         i++) {  // use 1st and 2nd 5th
-      ped += charge[i];
+                        //	for (int i = pedLength; i < 3*pedLength ; i++) {
+                        ////use 2nd and third 5th
+    if (charge.size() > 8) {
+      if (verbose_) ldmx_log(debug) << "going into oscillations check ";
+      for (int i = 3; i < charge.size() - 4; i++) {
```

Comment:
Can you add a comment on why we are skipping the first 2? is this supposed to be `pedOffset+1`? If yes, I'd prefer to move to a named constant than just a the hardcoded number